### ISSUE TITLE
when using fp16 - loss is going to be NaN and a lot of errors "OVERFLOW"

### ISSUE THREAD

**Issue by exelents** [microsoft/DeepSpeed#643]
I'm trying to use ZeRO optimizer with cpu offload mode, but it can work only with fp16 mode. When I turn on fp16 mode in configuration file and run command, I receive loss value equals to NaN in almost all steps. Also I get an error every step, like this:

`[2021-01-07 20:10:55,863] [INFO] [stage2.py:1361:step] [deepscale] OVERFLOW! Rank 0 Skipping step. Attempted loss scale: 64.0, reducing to 32.0`

Model which I'd like to fine tune is T5 large (have plans to try 3b and 11b ones). Using it through Huggingface transformers library. Here is my fine-tuning code:
https://gist.github.com/exelents/ebbb9b0e560d9b916532a0adce16bbcb
and here is config:
https://gist.github.com/exelents/3a2e057f4503413ed8b3b2570f650c15

Can anybody help, pointing what I'm doing wrong?

**Comment by exelents** [comment]
Guys? I've tried examples from DeepSpeedExamples repo, used a BingBert training code, and replaced Bert on T5 there. Here a result code:
https://github.com/exelents/try_t5
How to run:
`./run-qa-2.sh 1 0 . ./output`
But even there I have the same troubles. Loss is going to be NaN and some(not all) tensors in output features are also filled with NaN.

Maybe it can be a problem with Huggingface library? BingBert from DeepSpeedExamples works well for me, so it's not my installed software problem...

If anybody would like to test this code - here is binary compiled dataset for this training:
https://drive.google.com/file/d/1oxCxYCuCWebmaUQ_s9il7EDBkisL7x-_/view?usp=sharing
https://drive.google.com/file/d/1WCzxAnp2bEllbQ0_2d_6hoq5tQjxBFXh/view?usp=sharing

**Comment by exelents** [comment]
I have solved my troubles compiling DeepSpeed with Cuda 11.1 and Nvidia computing platform 8.6 compatibility.
